feat: allow opening hidden plugins

alvaro-cuesta opened the pull request:
Please tick as appropriate:
- [x] I have tested this code on a steam deck or on a PC
- [x] My changes generate no new errors/warnings
- [ ] This is a bugfix/hotfix
- [x] This is a new feature

# Description

This fixes issue: #734 (see issue for alternatives considered)

This solves the use case where plugins are hidden from the quick access list because access to their panels is not used very often. For example, plugin "ProtonDB Badges" is fire-configure-once-and-forget, so it makes sense to hide it so it doesn't waste space and attention during quick access.

But this doesn't mean the user will never want to access hidden plugin panels to e.g. briefly change configuration. Currently the only way to access the plugin content panel is to unhide the plugin, access it in the quick access list, and then hide it again.

This allows opening plugins from the "Plugins" panel in Decky's settings, even if the plugin is hidden in the quick access list. Clicking the open button opens the plugin in the quick access sidebar, regardless of its hidden status.

Additionally:

- Since the "open" action is expected to be more common than the current "Reinstall" button (which seems to be there because it turns into "Update" when an update is available for the plugin):

    - Add "Open" action as the primary icon button
    - Move "Reinstall" option inside context menu so plugin can always be reinstalled (regardless of available updates)
    - Keep "Update" button in main interactables, but only show it when an update is available

    _(See [my comment below](https://github.com/SteamDeckHomebrew/decky-loader/pull/736#issuecomment-2564306625) for some thoughts around this.)_

- Add notification badge if plugin has updates

    Makes it more consistent with the rest of the app, which already has badges on available updates. It also makes it more obvious now that the "Update" button is only shown conditionally, calling attention to the updatable entries.

This is how the changes look in the "Plugins" settings tab:

![Screenshot](https://github.com/user-attachments/assets/39cf7071-1847-4c52-8953-bafd312c7f9a)

alvaro-cuesta committed:
feat: add "open" action to plugins settings

This allows opening plugins even if they are hidden in the quick access
list.

This solves the use case where plugins are hidden from the quick access
list because access to their panels is not used very often. For example,
plugin "ProtonDB Badges" is fire-configure-once-and-forget, so it makes
sense to hide it so it doesn't waste space and attention.

But this doesn't mean the user will never want to access hidden plugin
panels to e.g. briefly change configuration. Currently the only way to
access the plugin content panel is to unhide the plugin, access it in
the quick access list, and then hide it again.

Resolves #734

alvaro-cuesta committed:
feat: button to open sidebar as main action

The action is expected to be more common than the current "Reinstall"
button (which seems to be there because it turns into "Update" when an
update is available for the plugin).

Instead:

- Remove "Open" action from context menu and add it as an icon button
- Move "Reinstall" option to context menu (regardless of available
  updates) so plugin can always be reinstalled
- Keep "Update" button in main interactables, but only show it when an
  update is available

alvaro-cuesta committed:
feat: add notification badge if plugin has updates

Makes it more consistent with the rest of the app, which already has
badges on available updates. It also makes it more obvious now that the
"Update" button is only shown conditionally, calling attention to the
updatable entries.

alvaro-cuesta commented:
Posted this PR because I'm running it on my fork and I figured having an actual runnable build would make sense to better discuss the ergonomics (see #734 for alternatives considered).

In particular, I'm not entirely sure if the "Open" action should be relegated to the context menu anyways, instead of having its own icon button.

I want to work on #350 soon-ish (which might deserve a toggle per row for conciseness, see [screenshot in first comment](https://github.com/SteamDeckHomebrew/decky-loader/issues/350#issuecomment-1399525121)) and want to leave a smooth path forward without too many UI changes, so it might make sense to not even add the "Open" as a main action in this PR to leave some space for the toggle.

- I expect 99% of the time users will just use the quick access menu, and for the very few cases that they really want to open a hidden plugin they can still reach through the context menu which is one click away.
- Seems like the toggle might be a useful action to have ready at hand, especially after updates that might break multiple plugins at once...
- ...but "open" seems useful as a main action too, so I can't make up my mind on which of the actions should stay (or both?)

I still think it makes sense to keep the "Reinstall" button tucked inside the context menu though (for the same space-saving reasons).

Some actual screenshots on how it could look like IF the plugins are disabled via toggle:

- This would be the "Plugins" tab with "Open" button AND disable toggle, which admittedly looks a bit busy and makes overflows more likely:

    ![With open button](https://github.com/user-attachments/assets/62c27473-b833-4ead-b9ae-37c43c1790b6)

- And this would be only the disable toggle, no "Open" button:

    ![No open button](https://github.com/user-attachments/assets/1e4db544-ebe6-433d-8e9c-024df2525fc8)

- If the open action is tucked inside the context menu it could look like this (note that this uses my branch of SteamDeckHomebrew/decky-frontend-lib#115 which exposes `MenuSeparator`):

    ![Context menu](https://github.com/user-attachments/assets/dca70f50-2483-4db4-8530-3d594344d2b0)

Thoughts?

alvaro-cuesta committed:
fix: remove unused new transalation

TrainDoctor commented:
@alvaro-cuesta, hey sorry for the long time between when we last discussed this on discord and now.
@EMERALD0874 is our most versed UI/UX person on the team. I'll ask them to give this a review. Personally I love the toggle and open button in combination but I want Emerald's opinion if that's too much visually. I'm aware that a large portion of our users don't even know to open the settings menu until a plugin starts misbehaving so adding more interface might not be helpful.
Is it possible to make the open button only appear on plugins that have been hidden? If so I think that would be ideal imo.

alvaro-cuesta commented:
@TrainDoctor no worries, holidays are always a tough time, thanks for coming back to this.

> Is it possible to make the open button only appear on plugins that have been hidden?

The UX person at my job has this motto of better being consistent with what you show. E.g. in this case it might not be immediately obvious why some plugins have the buttons while others don't, which might puzzle some users.

But of course there are always exceptions and although I like dabbling in UX I'm not an expert either, let's see what @EMERALD0874 thinks!

alvaro-cuesta committed:
feat: add menu separator to plugin menu

Splits it in two sections:

- Install actions (reload, reinstall, uninstall)
- Status actions (hide, freeze)

Makes a bit less overwhelming and more structured, now that there is an
additional "Reinstall" menu item.

alvaro-cuesta commented:
Now that SteamDeckHomebrew/decky-frontend-lib#115 is merged I added the menu separator to the plugin's menu (tested in both SteamOS Stable and Beta), splitting it in two sections:

- Install actions (reload, reinstall, uninstall)
- Status actions (hide, freeze)

Makes a bit less overwhelming and more structured, since this PR adds an additional menu item (and possibly more to come) with the "Reinstall" action, making the plugin list a bit more crowded. IMO it makes the context menu a bit more clear.

Normal mode:

![Normal mode screenshot](https://github.com/user-attachments/assets/f50370dc-e5d3-4038-aa7c-96f5b0061e5d)

Developer mode:

![Developer mode screenshot](https://github.com/user-attachments/assets/faeea034-51b5-443d-8002-c788faaf7fea)

Even though the "normal" mode only has one status item I still think it's a good idea to guide the user through both sets of actions, plus SteamOS itself has multiple instances of guiding like this, e.g. see how "Switch to Desktop" is standing alone here to highlight it's its own kinda thing:

![Power menu screenshot](https://github.com/user-attachments/assets/a91167ec-cf43-4ca8-8e81-b68a547fcdf2)

EMERALD0874 commented:
@alvaro-cuesta @TrainDoctor -  I'm sorry that I didn't see this until now. I wasn't requested as a reviewer and I think I have some weird settings for this repo with mentions, so it went under my radar.

I think both showing the plugin in the QAM and toggling them on/off should be put within the context menu, not a toggle. I imagine most users will want to have the plugins they installed consistently on, so making it possible to accidentally turn them off could be problematic compared to a button labeled "Enable" or "Disable" depending on the context. Maybe that could go beneath "Reload"?
